fms/models/llama.py

```diff
@@ -324,11 +325,15 @@ def forward(
         if only_last_token:
             output = output[:, -1, :]
         preds = self.shared(output, reverse=True)
-
+        
+        out = [preds]
         if use_cache:
-            return preds, cache
-        else:
-            return preds
+            out.append(cache)
+        if include_embeds:
+            out.append(output)
+        if len(out)==1:
+            return out[0]
+        return out
```

Comment:
if we're changing the outputs of a model, we should either have all models doing the same, (aka gpt bigcode, roberta) or at least document the difference somewhere

Reply:
True... I need some way to access the embeddings of the model and not just the logits. Whatever way is most appropriate here, I can then propagate to other (generative) models

Reply:
IIUC this is a common use-case we plan to handle via a "headless" model instance rather than an extra parameter. I think this was required in HF and motivated the separate `_helper` function. @JRosenkranz probably has a view on how best to structure that (across all of our models)